src/mui/input/CheckboxGroupInput.js

```diff
@@ -0,0 +1,123 @@
+
+/**
+ * An Input component for a checkbox group, using an array of objects for the options
+ *
+ * Pass possible options as an array of objects in the 'choices' attribute.
+ *
+ * By default, the options are built from:
+ *  - the 'id' property as the option value,
+ *  - the 'name' property an the option text
+ * @example
+ * const choices = [
+ *    { id: 'M', name: 'Male' },
+ *    { id: 'F', name: 'Female' },
+ * ];
+ * <CheckboxGroupInput source="gender" choices={choices} />
+ *
+ * You can also customize the properties to use for the option name and value,
+ * thanks to the 'optionText' and 'optionValue' attributes.
+ * @example
+ * const choices = [
+ *    { _id: 123, full_name: 'Leo Tolstoi', sex: 'M' },
+ *    { _id: 456, full_name: 'Jane Austen', sex: 'F' },
+ * ];
+ * <CheckboxGroupInput source="author_id" choices={choices} optionText="full_name" optionValue="_id" />
+ *
+ * `optionText` also accepts a function, so you can shape the option text at will:
+ * @example
+ * const choices = [
+ *    { id: 123, first_name: 'Leo', last_name: 'Tolstoi' },
+ *    { id: 456, first_name: 'Jane', last_name: 'Austen' },
+ * ];
+ * const optionRenderer = choice => `${choice.first_name} ${choice.last_name}`;
+ * <CheckboxGroupInput source="author_id" choices={choices} optionText={optionRenderer} />
+ *
+ * `optionText` also accepts a React Element, that will be cloned and receive
+ * the related choice as the `record` prop. You can use Field components there.
+ * @example
+ * const choices = [
+ *    { id: 123, first_name: 'Leo', last_name: 'Tolstoi' },
+ *    { id: 456, first_name: 'Jane', last_name: 'Austen' },
+ * ];
+ * const FullNameField = ({ record }) => <span>{record.first_name} {record.last_name}</span>;
+ * <CheckboxGroupInput source="gender" choices={choices} optionText={<FullNameField />}/>
+ *
+ * The object passed as `options` props is passed to the material-ui <RadioButtonGroup> component
+ */
+export class CheckboxGroupInput extends Component {
+    handleChange = (event, value) => {
+        console.log(this.props.input.value);
+        this.props.input.onChange(value);
+    }
+
+    render() {
+        const { input, label, source, choices, optionText, optionValue } = this.props;
+        const option = React.isValidElement(optionText) ? // eslint-disable-line no-nested-ternary
+            choice => React.cloneElement(optionText, { record: choice }) :
+            (typeof optionText === 'function' ?
+                optionText :
+                choice => choice[optionText]
+            );
+
+        return (
+            <Labeled label={label} onChange={this.handleChange} source={source}>
+                <div>
+                    {choices.map((choice, index) => (
+                        <input
+                            key={choice[optionValue]}
+                            type="checkbox"
+                            checked={input.value.indexOf(choice[optionValue]) >= 0}
+                            onChange={event => {
```

Comment:
move that to a property function, otherwise it forces rerender as it's always a new function